### PR TITLE
Designing the Quran Home Screen

### PR DESCRIPTION
**Pull request by rezaiyan**
In following of [QuranHome ](#14)PR, to Quran Home design,
I'm separating `Surah` & `Juz` list into two pages of a ViewPager, in consequence, I decided to separate the entire logic of each page (`Screen`).

This is a preview of what we want:

### CODE REVIEW

**Comment by rezaiyan** [comment]
There is a little untidy,
hold on! 
:)

File: buildSrc/src/main/java/Dependencies.kt [hunk 1]
```diff
@@ -104,7 +104,8 @@ object Deps {
             val timber = "com.jakewharton.timber:timber:4.7.1"
             val fresco = "com.facebook.fresco:fresco:${Versions.fresco}"
             val frescoOkHttp = "com.facebook.fresco:imagepipeline-okhttp3:${Versions.fresco}"
-            val conductor = "com.bluelinelabs:conductor:3.0.0-rc2"
+            val conductor = "com.bluelinelabs:conductor:3.0.0-rc3"
```
Comment (helloseyedjafari) [inline buildSrc/src/main/java/Dependencies.kt]: let's extract the versions

File: buildSrc/src/main/java/Dependencies.kt [hunk 2]
```diff
@@ -8,7 +8,7 @@ object Versions {
     val supportJunitExt = "1.0.0"
     val androidSupport = "1.0.0"
     val googleMaterial = "1.0.0-rc01"
-    val constraintLayout = "2.0.0-beta4"
+    val constraintLayout = "2.0.0-beta3"
```
Comment (helloseyedjafari) [inline buildSrc/src/main/java/Dependencies.kt]: stable 2 is released

File: quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/home/HomePagerAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import com.bluelinelabs.conductor.Controller
+import com.bluelinelabs.conductor.Router
+import com.bluelinelabs.conductor.RouterTransaction
+import com.bluelinelabs.conductor.support.RouterPagerAdapter
+import com.islamversity.quran_home.feature.juz.JuzListView
+import com.islamversity.quran_home.feature.surah.SurahListView
+
+/**
+ * @author Ali (alirezaiyann@gmail.com)
```
Comment (helloseyedjafari) [inline quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/home/HomePagerAdapter.kt]: initials

File: quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/home/HomePagerAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                1 -> router.setRoot(RouterTransaction.with(JuzListView()))
+            }
+        }
+    }
+
+    override fun getCount() = 2
+
+    override fun getPageTitle(position: Int): CharSequence? {
+        return when (position) {
+            0 -> "سوره"
```
Comment (helloseyedjafari) [inline quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/home/HomePagerAdapter.kt]: resources please

File: quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/home/QuranHomeView.kt [hunk 5]
```diff
@@ -1,30 +1,31 @@
...
 import javax.inject.Inject
-import kotlin.time.DurationUnit
 import kotlin.time.ExperimentalTime
-import kotlin.time.toDuration
 
 
 class QuranHomeView : CoroutineView<QuranHomeViewBinding, QuranHomeState, QuranHomeIntent>() {
 
     private val intentChannel = Channel<QuranHomeIntent>(Channel.UNLIMITED)
+    private val pagerAdapter: HomePagerAdapter by lazy {
```
Comment (helloseyedjafari) [inline quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/home/QuranHomeView.kt]: Do we really need the `Lazy`?

File: quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/juz/JuzListView.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import com.islamversity.quran_home.feature.juz.di.DaggerJuzListComponent
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.merge
+import kotlinx.coroutines.flow.receiveAsFlow
+import javax.inject.Inject
+
+/**
+ * @author Ali (alirezaiyann@gmail.com)
```
Comment (helloseyedjafari) [inline quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/juz/JuzListView.kt]: Initials

File: quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/surah/SurahListView.kt [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import com.islamversity.quran_home.feature.surah.di.DaggerSurahListComponent
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.merge
+import kotlinx.coroutines.flow.receiveAsFlow
+import javax.inject.Inject
+
+/**
+ * @author Ali (alirezaiyann@gmail.com)
```
Comment (helloseyedjafari) [inline quranHome/src/androidMain/Kotlin/com/islamversity/quran_home/feature/surah/SurahListView.kt]: initials

**Review by helloseyedjafari — APPROVED** [review]
well done

**Review by helloseyedjafari — APPROVED** [review]

**Pull request merged**